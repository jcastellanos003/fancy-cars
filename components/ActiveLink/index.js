import { useRouter } from 'next/router'
import Link from 'next/link';

const ActiveLink = ({ children, ...props }) => {
  const router = useRouter();
  const child = React.Children.only(children);
  
  return (
    <Link {...props}>
      {React.cloneElement(child, { active: router.pathname === props.href, theme: props.theme })}
    </Link>
  )
};

export default ActiveLink;
