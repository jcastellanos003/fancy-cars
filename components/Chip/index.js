import PropTypes from 'prop-types';

import * as components from './styles';

export const Chip = ({ name, onClose }) => (
  <components.StyledChip>
    <span>{name}</span> <components.CloseIcon onClick={onClose} />
  </components.StyledChip>
);

Chip.propTypes = {
  name: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired
};
