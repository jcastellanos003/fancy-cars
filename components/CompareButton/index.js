import * as components from './styles';

export const CompareButton = () => (
  <components.StyledCompareButton>
    <components.CompareButtonIcon />
  </components.StyledCompareButton>
);
