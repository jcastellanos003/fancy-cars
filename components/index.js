export * from './ActiveLink';
export * from './Nav';
export * from './Meta';
export * from './SearchBox';
export * from './Card';
export * from './VehicleCard';
export * from './BackButton';
export * from './PageHeader';
export * from './Chip';
export * from './CompareButton';
export * from './PageTitle';
export * from './PriceLabel';
