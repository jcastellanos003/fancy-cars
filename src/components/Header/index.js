import { Nav } from '../Nav';
import * as styled from './styles';

export const Header = () => (
  <section>

    <styled.CustomHeader>
      <styled.Logo src="/logo.png"/>
    </styled.CustomHeader>

    <Nav />

  </section>
);
