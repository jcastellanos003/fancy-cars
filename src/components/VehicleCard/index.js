import PropTypes from 'prop-types';

import { Card, CardHeader, CardContent  } from '../Card';
import { VehicleImage } from './styles';

export const VehicleCard = ({
    image,
    size,
    children
  }) => (
    <section>
      <Card size={size}>

        <CardHeader>
          <VehicleImage src={image} />
        </CardHeader>

        <CardContent>
          {children}
        </CardContent>

      </Card>
    </section>
);

VehicleCard.propTypes = {
  size: PropTypes.oneOf(['medium', 'large']).isRequired,
  image: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
