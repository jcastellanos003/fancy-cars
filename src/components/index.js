export * from './ActiveLink/ActiveLink';
export * from './BackButton/BackButton';
export * from './Card';
export * from './CompareButton/CompareButton';
export * from './CompareLink/CompareLink';
export * from './Chip/Chip';
export * from './Header/Header';
export * from './HorizontalPane/HorizontalPane';
export * from './InnerPage/InnerPage';
export * from './Master/Master';
export * from './Meta/Meta';
export * from './Nav/Nav';
export * from './PageHeader/PageHeader';
export * from './PageTitle/PageTitle';
export * from './PriceLabel/PriceLabel';
export * from './SearchBox/SearchBox';
export * from './SelectedCard/SelectedCard';
export * from './SelectedButton/SelectedButton';
export * from './VehicleCard/VehicleCard';
