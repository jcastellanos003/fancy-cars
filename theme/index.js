export * from './DefaultTheme';
export * from './GlobalStyles';
