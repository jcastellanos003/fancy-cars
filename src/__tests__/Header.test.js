import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { Header } from '../components';
import { theme } from '../theme';

describe('Header Component', () => {
  test('Matches the snapshot', () => {
    const header = create(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>
      );

    expect(header.toJSON()).toMatchSnapshot();
  })
})
