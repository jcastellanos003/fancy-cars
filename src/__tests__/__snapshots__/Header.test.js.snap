// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Matches the snapshot 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  margin-right: 0.5rem;
  width: 1.3rem;
}

.c5 {
  margin-right: 0.5rem;
  width: 1.3rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  cursor: pointer;
  background: #2A293B;
  border-bottom: 5px solid #201f2f;
  color: #FAFAFA;
  font-size: 0.8rem;
  font-weight: 500;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  -webkit-transition: color 0.5s;
  transition: color 0.5s;
}

.c3:hover {
  color: #FFC82C;
}

.c3:hover .icon {
  color: #FFC82C;
}

.c3 .icon {
  color: #FAFAFA;
  -webkit-transition: color 0.5s;
  transition: color 0.5s;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #2A293B;
  height: 50px;
}

.c1 {
  width: 10rem;
  padding-top: 0.2rem;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #2A293B;
}

<section>
  <header
    className="c0"
  >
    <img
      className="c1"
      src="/logo.png"
    />
  </header>
  <div
    className="c2"
  >
    <a
      className="c3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <svg
        aria-hidden="true"
        className="c4 c5 icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      Home
    </a>
    <a
      className="c3"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <svg
        aria-hidden="true"
        className="c4 c6 icon"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <path
            d="M0 0h24v24H0V0z"
          />
        </defs>
        <clipPath>
          <use
            overflow="visible"
            xlinkHref="#a"
          />
        </clipPath>
        <path
          clipPath="url(#b)"
          d="M9.01 14H2v2h7.01v3L13 15l-3.99-4v3zm5.98-1v-3H22V8h-7.01V5L11 9l3.99 4z"
        />
      </svg>
      Compare
    </a>
  </div>
</section>
`;
